projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py

```diff
@@ -0,0 +1,193 @@
+        angle_left = (angle_left + 2 * np.pi) % (2 * np.pi)
+        if (angle_left > np.pi):
+            angle_left -= 2 * np.pi
+
+        # Turn the robot to the left or to the right according the value and the sign of angle_left
+        yaw_disturbance = self.MAX_YAW_DISTURBANCE * angle_left / (2 * np.pi)
+        # non proportional and decreasing function
+        pitch_disturbance = clamp(
+            np.log10(abs(angle_left)), self.MAX_PITCH_DISTURBANCE, 0.1)
+
+        if verbose_movement:
+            distance_left = np.sqrt(((self.target_position[0] - self.current_pose[0]) ** 2) + (
+                (self.target_position[1] - self.current_pose[1]) ** 2))
+            print("remaning angle: {:.4f}, remaning distance: {:.4f}".format(
+                angle_left, distance_left))
+        return yaw_disturbance, pitch_disturbance
+
+    def run(self):
+        t1 = self.getTime()
+
+        roll_disturbance = 0
+        pitch_disturbance = 0
+        yaw_disturbance = 0
+
+        # Specify the patrol coordinates
+        waypoints = [[-30, 20], [-60, 20],  [-60, 10], [-30, 5]]
+        # target altitude of the robot in meters
+        self.target_altitude = 15
+
+        while self.step(self.time_step) != -1:
+
+            # Read sensors
+            roll, pitch, yaw = self.imu.getRollPitchYaw()
+            Xpos, Ypos, altitude = self.gps.getValues()
+            roll_acceleration, pitch_acceleration, _ = self.gyro.getValues()
+            self.set_position([Xpos, Ypos, altitude, roll, pitch, yaw])
+
+            if altitude > self.target_altitude - 1:
+                # as soon as it reach the target altitude, compute the disturbances to go to the given waypoints.
+                if self.getTime() - t1 > 0.1:
+                    yaw_disturbance, pitch_disturbance = self.move_to_target(
+                        waypoints)
+                    t1 = self.getTime()
+
+            roll_input = self.K_ROLL_P * \
+                clamp(roll, -1, 1) + roll_acceleration + roll_disturbance
+            pitch_input = self.K_PITCH_P * \
+                clamp(pitch, -1, 1) + pitch_acceleration + pitch_disturbance
+            yaw_input = yaw_disturbance
+            clamped_difference_altitude = clamp(
+                self.target_altitude - altitude + self.K_VERTICAL_OFFSET, -1, 1)
+            vertical_input = self.K_VERTICAL_P * \
+                pow(clamped_difference_altitude, 3.0)
+
+            front_left_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input - yaw_input + pitch_input - roll_input
+            front_right_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input + yaw_input + pitch_input + roll_input
+            rear_left_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input + yaw_input - pitch_input - roll_input
+            rear_right_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input - yaw_input - pitch_input + roll_input
```

Comment:
According to the our python coding style we can have up to 128 characters per line, it shouldn't be necessary to write most of this in two lines (from L165 to L182)